Aksarantara/pegon-ocr-integration
# feature/api-key-auth

## Description
This PR adds API key authentication to the `/infer` endpoint in the FastAPI application. It ensures that the endpoint can only be accessed with a valid API key, providing an extra layer of security. Additionally, the requirements.txt file has been updated to include all the necessary dependencies required for the application.

## Changes Made
- Added API key authentication mechanism to the `/infer` endpoint.
- Updated the requirements.txt file with the necessary dependencies.

## How to Test
1. Run the FastAPI application.
2. Use `curl` or any other HTTP client to send a `POST` request to the `/infer` endpoint with a valid API key.
3. Verify that the request is successful and the endpoint returns the expected results.
4. Try accessing the `/infer` endpoint without providing a valid API key and verify that the request is rejected with a 401 Unauthorized response.

Please review and merge this PR once it's ready
